pkg/kine/logstructured/sqllog/sql.go

```diff
@@ -18,8 +18,9 @@ import (
 )
 
 const (
-	SupersededCount = 100
-	otelName        = "sqllog"
+	SupersededCount  = 100
+	compactBatchSize = 1000
```

Comment:
That's a great point @bschimke95 
IMHO we should be fine with this default batch size. Note that kine does one big transaction here instead of batches which can block a db. These batches address those high workload scenarios and 1000 revisions should still be comfortably manageable under high load. 

Nice-to-know:
I can't find anywhere upstream etcd is setting a compaction batch size, but they also have some sort of logic in place where they can stop doing compaction under high load. I think this comes from the etcdserver request and we are not compacting on this request but simply every 5 minutes or whatever our compaction interval is. If I remember correctly, the compaction interval can be tweaked (the time between each compaction). 
This could be worth investigating as part of future work.

Overall, if we do run into issues under high load, I think we will need a change that goes beyond changing the batch size.

Reply:
cool, thanks for the context